driver-core/src/main/com/mongodb/internal/authentication/SaslPrep.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.authentication;
+
+import java.nio.CharBuffer;
+import java.text.Normalizer;
+
+/**
+ * Utility class for Sasl string preparation.
+ *
+ * <p>This class should not be considered a part of the public API.</p>
+ */
+public final class SaslPrep {
```

Comment:
What's the github project where this came from?

Reply:
https://github.com/tombentley/saslprep

Reply:
thanks.  just wanted to see if it had any stars/usage